hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/internal/EmbeddableTypeImpl.java

```diff
@@ -39,4 +41,18 @@ public EmbeddableTypeImpl(
 	public PersistenceType getPersistenceType() {
 		return PersistenceType.EMBEDDABLE;
 	}
+
+	public int getTupleLength() {
+		int count = 0;
+		for ( SingularAttribute<? super J, ?> attribute : getSingularAttributes() ) {
+			final Type<?> type = attribute.getType();
+			if ( type.getPersistenceType() == PersistenceType.EMBEDDABLE ) {
+				count += ( (DomainType<?>) type ).getTupleLength();
+			}
+			else {
+				count += 1;
+			}
```

Comment:
Minor simplification suggestion.

```suggestion
			count += ( (DomainType<?>) attribute.getType() ).getTupleLength();
```

Reply:
Sure, this works.